src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +148,221 @@ async function uploadWithProgress({
+                return undefined
+            }
+        }
+    } else {
+        return response.bucket
+    }
+    return undefined
+}
+
+export async function uploadFilePaletteCommand(
+    fileSizeBytes: FileSizeBytes = statFile,
+    window = Window.vscode(),
+    outputChannel = ext.outputChannel
+): Promise<void>{
+
+    //1 step: get the file that is currently opened in the editor
+    const activeEditor: TextEditor | undefined = vscode.window.activeTextEditor // returns an object of type TextEditor | undef 
+    const document = activeEditor?.document
+    let selectAnotherFile = false
+    let fileLocation: vscode.Uri | undefined
+
+
+    //2 step: prompt the quick picker so that user can specify if they want the
+    //current file or select other file from their system
+    //there is 2 possibilities
+    //there is an active editor with a file (document) open, then show the quickpick with the option 
+    //to browse for more files
+    //or there is no active editor, no file is open
+    if (!document) {
+        fileLocation = await promptForFileLocation(Window.vscode())
+    } 
+    else {
+        fileLocation = document.uri
+        const fileNameToDisplay = path.basename(fileLocation.fsPath)
+
+        //Now, ask the user with quickpicker to select file to use
+
+        //crete quick pick using picker.createQuickPick(options, items, buttons) that returns a QuickPick
+        const fileOption : vscode.QuickPickItem = { 
+            label: addCodiconToString('file', fileNameToDisplay) //TODO:: change parameters to correct ones
+        }
+        const selectMore: vscode.QuickPickItem = {
+            label: "Browse for more files..."
+        }
+
+        const picker = createQuickPick({
+            options: {
+                canPickMany: false,
+                ignoreFocusOut: true,
+                title: localize('AWS.message.selectFileUpload', 'Couldnt find it'), //not finding this wtf
+                step: 1,
+                totalSteps: 2,
+            },
+            items: [fileOption, selectMore],
+            buttons: [vscode.QuickInputButtons.Back]
+        })
+
+        //use the quickPick item above with promptUser(picker, onDidtTriggerButton) to show the quickpick
+        const response = verifySinglePickerOutput(await promptUser({
+            picker: picker,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                }
+            },
+        }))
+
+        if(!response){
+            getLogger().info('UploadFile cancelled')
+            telemetry.recordS3UploadObject({ result: 'Cancelled' })
+            return
+        }
+
+        if (response!.label === 'Browse for more files...') {
+            selectAnotherFile = true
+        }
+
+    //2.1 step: if user wants other file prompt the file selector to allow user to select wanted file
+        if (selectAnotherFile) {
```

Comment:
Are there any other ways `selectAnotherFile` can be set to true? If not, remove`selectAnotherFile` entirely and move the logic in this if statement to line 316.